velox/dwio/parquet/common/BloomFilter.h

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Adapted from Apache Arrow.
+
+#pragma once
+
+#include "Hasher.h"
+#include "velox/common/base/BitUtil.h"
+#include "velox/dwio/common/BufferedInput.h"
+#include "velox/dwio/common/OutputStream.h"
+#include "velox/dwio/parquet/thrift/ParquetThriftTypes.h"
+
+#include <cmath>
+#include <cstdint>
+#include <memory>
+
+namespace facebook::velox::parquet {
+
+// A Bloom filter is a compact structure to indicate whether an item is not in a
+// set or probably in a set. The Bloom filter usually consists of a bit set that
+// represents a set of elements, a hash strategy and a Bloom filter algorithm.
+class BloomFilter {
+ public:
+  // Maximum Bloom filter size, it sets to HDFS default block size 128MB
+  // This value will be reconsidered when implementing Bloom filter producer.
+  static constexpr uint32_t kMaximumBloomFilterBytes = 128 * 1024 * 1024;
+
+  /// Determine whether an element exist in set or not.
+  ///
+  /// @param hash the element to contain.
+  /// @return false if value is definitely not in set, and true means PROBABLY
+  /// in set.
+  virtual bool findHash(uint64_t hash) const = 0;
```

Comment:
Making this virtual just destroys the performance.  Maybe ok since this is only applied at row group level for now.  But as soon as we need to apply it on row-wise data, this will need to be revamped.

Reply:
In C++ Part we expect `insertHashes` to be used

Besides using BlockSplitBloomFilter directly is also ok since Parquet only support this kind of bloom filter for now